lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -160,6 +161,13 @@ public static function triggerIfCalledFromOutside(string $package, string $link,
         self::delegateTriggerToBackend($message, $backtrace, $link, $package);
     }
 
+    private static function logger(): LoggerInterface
+    {
+        assert(self::$logger !== null);
```

Comment:
I would actually put that inline inside the `if (($type & self::TYPE_PSR_LOGGER) > 0) {`. This invariant is true only if this check on the type is performed